packages/office-ui-fabric-react/src/components/SwatchColorPicker/ColorPickerGridCell.base.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import {
+  autobind,
+  customizable
+} from '../../Utilities';
+import {
+  IColorCellProps,
+  IColorPickerGridCellProps,
+  IColorPickerGridCellStyleProps,
+  IColorPickerGridCellStyles
+} from './ColorPickerGridCell.types';
+import { getColorFromString } from '../../utilities/color/colors';
+import { GridCell } from '../../utilities/grid/GridCell';
+import { IGridCellProps } from '../../utilities/grid/GridCell.types';
+import { classNamesFunction, IClassNames } from '../../Styling';
+
+const getClassNames = classNamesFunction<IColorPickerGridCellStyleProps, IColorPickerGridCellStyles>();
+
+class ColorCell extends GridCell<IColorCellProps, IGridCellProps<IColorCellProps>> {
+}
+
+@customizable('ColorPickerGridCell', ['theme'])
+export class ColorPickerGridCellBase extends React.Component<IColorPickerGridCellProps, {}> {
+
+  public static defaultProps = {
+    circle: true,
+    disabled: false,
+    selected: false,
+  } as IColorPickerGridCellProps;
+
+  private _classNames: {[key in keyof IColorPickerGridCellStyles]: string };
+
+  public render() {
+    let {
+      item,
+      id,
+      selected,
+      disabled,
+      getStyles,
+      theme,
+      circle
+    } = this.props;
+
+    this._classNames = getClassNames(
+      getStyles!,
+      {
+        theme: theme!,
+        disabled,
+        selected,
+        circle
+      }
+    );
+
+    return (
+      <ColorCell
+        item={ item }
+        id={ id }
+        key={ item.id }
+        disabled={ disabled }
+        role={ 'gridcell' }
+        onRenderItem={ this._onRenderColorOption }
+        selected={ selected }
+        onClick={ this.props.onClick }
+        onHover={ this.props.onHover }
+        onFocus={ this.props.onFocus }
+        label={ item.label }
+        className={ this._classNames.colorCell }
+        classNames={ this._classNames }
+      />
+    );
+  }
+
+  /**
+ * Render the core of a color cell
+ * @returns {JSX.Element} - Element representing the core of the item
+ */
+  @autobind
+  private _onRenderColorOption(colorOption: IColorCellProps): JSX.Element {
+    // Build an SVG for the cell with the given shape and color properties
+    return (
+      <svg className={ this._classNames.svg } viewBox='0 0 20 20' fill={ getColorFromString(colorOption.color as string)!.str } >
```

Comment:
This as string typecast looks weird to me. Was that already there? I wonder why the color is optional...

Reply:
Yes this was already there. Not sure why the color is optional.. I can just make it required or was there some reasoning behind it @jspurlin

Reply:
Okay we need the typecast because getcolorFromString returns IColor. I can make the color prop not optional though :)

Reply:
It's techincally a breaking change, but if having color be optional makes sense, we could probably make that change.

Reply:
I can't think of any reason why the color should be option

Reply:
Ah, looking at the code, it was because the SwatchColorPicker used to include the whole menu (headers, dividers, colorCells, and normal menuItems). I'm good with making it not optional